tests/test_database.py

```diff
@@ -273,3 +285,214 @@ def test_serviceability(self, reserved, requested_platform, requested_machine, i
         task.tags = [Tag("tag1")]
         # tasks matching the available machines are serviceable
         assert self.d.is_serviceable(task) is is_serviceable
+
+    @pytest.mark.parametrize(
+        "task_instructions,machine_instructions,expected_results,function",
+        (
+            #Assign 10 tasks to 10 specific machines availables
+            ([("tag1",10)],
+             [("windows","x64","tag1",10),("windows","x86","tag2",5),("linux","x64","tag3",2)],
+             {"tag1":10},
+             "db_relevant_machines_to_tasks"),
+            #Assign 10 tasks to 10 specific machines availables
+            ([("tag1",8),("tag2",2)],
+             [("windows","x64","tag1,",10),("windows","x86","tag2,",2),("linux","x64","tag3,",2)],
```

Comment:
Let's keep things simple as we ramp up the test cases
```suggestion
             [("windows","x64","tag1,",8),("windows","x86","tag2,",2)],
```